File: API/Entities/Product.cs

using System.Collections.Generic;

namespace API.Entities
{
    public class Product : BaseEntity
    {
        public string PartNumber { get; set; }
        public string Description { get; set; }

        public int BrandId { get; set; }
        public Brand Brand { get; set; }

        public int CategoryId { get; set; }
        public Category Category { get; set; }

        public double ItemVolume { get; set; }
        public double ItemWeight { get; set; }
        public int ItemPerSet { get; set; }
        public int Order { get; set; }


    }

    

    
    
     



}

File: client/src/app/models/product.ts

export interface Product {
    id: number;
    title: string;
    partNumber: string;
    description: string;
    brand: string;
    category: string;
    itemVolume: number;
    itemWeight: number;
    itemPerSet: number;
    order: number;

    pictureUrl : string;

}

export interface ProductParams {
    orderBy: string;
    searchTerm?: string;
    types?: string[];
    brands?: string[];
    pageNumber: number;
    pageSize: number;
}